cargo-pgx/src/main.rs

```diff
@@ -58,20 +58,22 @@ fn do_it() -> std::result::Result<(), std::io::Error> {
                     init_pgx(&Pgx::default(SUPPORTED_MAJOR_VERSIONS)?)
                 } else {
                     // user specified arguments, so we'll only install those versions of Postgres
-                    let default_pgx = Pgx::default(SUPPORTED_MAJOR_VERSIONS)?;
+                    let mut default_pgx = None;
                     let mut pgx = Pgx::new();
 
-                    for pg_config in versions.into_iter().map(|(pgver, pg_config_path)| {
-                        if pg_config_path == "download" {
-                            default_pgx
+                    for (pgver, pg_config_path) in versions {
+                        let config = if pg_config_path == "download" {
+                            if default_pgx.is_none() {
```

Comment:
Wanna do .get_or_insert() here instead?

Reply:
I initially did that too! However it ends up leaving the code more complicated, since you need to deal with the error from `Pgx::default(...)?`.

Reply:
Of course.  LGTM as-is.